docs/source/etcd-data-model.rst

```diff
@@ -339,23 +339,49 @@ Each rule sub-object has the following JSON-encoded structure:
 .. code-block:: json
 
     {
-      "protocol": "tcp|udp|icmp|icmpv6",
+      # Positive matches:
+      "protocol": "tcp | udp | icmp | icmpv6",
       "src_tag": "<tag_name>",
+      "src_selector": "<selector expression>",
       "src_net": "<CIDR>",
       "src_ports": [1234, "2048:4000"],
       "dst_tag": "<tag_name>",
       "dst_net": "<CIDR>",
       "dst_ports": [1234, "2048:4000"],
-      "icmp_type": <int>,
-      "action": "deny|allow",
+      "icmp_type": <int>, "icmp_code": <int>,  # Treated together, see below.
+
+      # Negated matches:
+      "!protocol": ...,
+      "!src_tag": ...,
+      "!src_selector": ...,
```

Comment:
@caseydavenport and I chatted about this.  We have mixed feelings on this:
- we both think that having !src_selector(...) be different to src_selector(!...) is unintuitive.
- we're not agreed on whether this behaviour is logically correct or not.
- we're not sure what we need it for anyway (I'm not entirely clear what the specific use case for this PR is at all tbh).

Therefore, can we please remove support for !src_selector?